client/src/panels/panel_declarations/results/IndicatorModalButton.js

```diff
@@ -0,0 +1,202 @@
+import gql from "graphql-tag";
+
+import { Fragment } from "react";
+
+import { withRouter } from "react-router";
+
+import {
+  Panel,
+  SpinnerWrapper,
+  StatelessModal,
+  WriteToClipboard,
+  create_text_maker_component,
+} from "../../../components/index.js";
+import { log_standard_event } from "../../../core/analytics.js";
+import { get_client } from "../../../graphql_utils/graphql_utils.js";
+
+
+import { IconCopyLink } from "../../../icons/icons.js";
+import { Indicator } from "../../../models/results.js";
+
+import { IndicatorDisplay } from "./result_components.js";
+import { text_maker as general_text_maker } from "./result_text_provider.js";
+
+
+import text from "./IndicatorModalButton.yaml";
+
+const { text_maker } = create_text_maker_component(text);
+
+const indicators_fields_fragment = `  id
+  stable_id
+  result_id
+  name
+  doc
+
+  target_year
+  target_month
+
+  target_type
+  target_min
+  target_max
+  target_narrative
+  measure
+  seeking_to
+
+  previous_year_target_type
+  previous_year_target_min
+  previous_year_target_max
+  previous_year_target_narrative
+  previous_year_measure
+  previous_year_seeking_to
+
+  target_explanation
+  result_explanation
+
+  actual_result
+  
+  status_key
+
+  methodology
+`;
+
+const get_indicator_query = gql`
+query($lang: String!, $id: String) {
+  root(lang: $lang) {
+    indicator(id: $id) {
+      ${indicators_fields_fragment}
+    }
+  }
+}
+`;
+
+const process_indicator = (indicator) => {
+  indicator.target_year = _.isNull(indicator.target_year)
+    ? null
+    : parseInt(indicator.target_year);
+  indicator.target_month = _.isNull(indicator.target_month)
+    ? null
+    : parseInt(indicator.target_month);
+  return indicator;
+};
+
+const query_api = (id) => {
+  const time_at_request = Date.now();
+  const client = get_client();
+  const query = get_indicator_query;
+  return client
+    .query({
+      query,
+      variables: {
+        lang: window.lang,
+        id,
+        _query_name: "results_bundle",
+      },
+    })
+    .then((response) => {
+      Indicator.create_and_register(
+        process_indicator(response.data.root.indicator)
+      );
+      return Promise.resolve();
+    })
+    .catch(function (error) {
+      const resp_time = Date.now() - time_at_request;
+      log_standard_event({
+        SUBAPP: window.location.hash.replace("#", ""),
+        MISC1: "API_QUERY_FAILURE",
+        MISC2: `Single indicator, took  ${resp_time} ms - ${error.toString()}`,
+      });
+      throw error;
+    });
+};
+
+class IndicatorModalButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { loading: true, show_modal: false };
+
+    const {
+      location: { search },
+      id,
+    } = this.props;
+
+    const query = new URLSearchParams(search).get("ind");
```

Comment:
`URLSearchParams` doesn't have IE11 support, not sure if we're polyfilling for it so this might be broken in IE.